### PR TITLE
421 health id number

### LINKED ISSUE #325: Make patient registration search result selectable
If a duplicate patient is found in the registration modal. The user should be about to select a patient and got to the patient's details page.

Improve working to make it clearer that the list shows existing patient and the user is about to create a potential duplicates. For example, change it something in the line "Similar patient(s) already exist, are you sure to create a new patient?"

### LINKED ISSUE #421: Store national id in names::national_health_number
mSupply actually has a national_health_number in the names table that can be used for the patient national id

### PR DESCRIPTION
**Pull request by clemens-msupply**
Wonder if the national_heath_number should be called `code_2` in omSupply which is a bit more generic...

- shows id and secondary id
- patient searchable by both ids
- now able to select matching patients from search results (#325)

Closes #421 
Closes #325

### CODE REVIEW

**Comment by mark-prins** [comment]
here's an unasked-for, and unrelated suggestion 😁 
The `Deceased` column would look neat as a checkbox:

Screen Shot 2022-08-08 at 4 20 37 PM

To do this.. I've added to the column, like so:
```
{
        key: 'isDeceased',
        label: 'label.deceased',
        align: ColumnAlign.Right,
        Cell: ReadOnlyCheckboxCell,
      },
```

and created a new component, the `ReadOnlyCheckboxCell` which implements the CheckboxCell, but forces the `isDisabled` prop:

```
import React from 'react';
import { RecordWithId } from '@common/types';
import { CellProps } from '../../../columns';
import { CheckboxCell } from './CheckboxCell';

export const ReadOnlyCheckboxCell = <T extends RecordWithId>({
  ...props
}: CellProps<T>): React.ReactElement<CellProps<T>> => (
  <CheckboxCell {...props} isDisabled />
);
```

File: client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx [hunk 1]
```diff
@@ -103,6 +114,9 @@ export const PatientResultsTab: FC<PatientPanel> = ({ patient, value }) => {
         data={data?.map(node => node.patient)}
         columns={columns}
         noDataMessage={t('messages.no-matching-patients')}
+        onRowClick={row => {
```
Comment (mark-prins) [inline client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx:117]: in using this - it was unclear what would happen when you click on the row.
I've put together an example of how a tooltip could be applied.. will make an issue to show that.

**Review by mark-prins — APPROVED** [review]
🎉  works well. not a big fan of `code2` as a property, it's too vague for me. the whole concept of multiple codes and identifiers is inherently vague though!

I have no better suggestions though.. 🤷

**Comment by clemens-msupply** [comment]
Thanks! I added the ReadOnlyCheckboxCell. Yes totally agree with code_2 but couldn't come up with a better solution neither. I could call it national_health_number to be consistent what is in the names table (but found this confusing as well because code is already the national id)?

Yes I think the search result tab needs better explanation, i.e. explain that the list shows possible duplicates. If no match is found we could also skip the whole tab?

**Comment by mark-prins** [comment]
> If no match is found we could also skip the whole tab?

Nice idea! or if not that, a friendly message saying 'there are no matching patients.. create one, happy in the knowledge that you won't be duplicating'

**Pull request merged**